src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -46,7 +46,7 @@ public function __construct(array $roles = array())
                 throw new \InvalidArgumentException(sprintf('$roles must be an array of strings, or RoleInterface instances, but got %s.', gettype($role)));
             }
 
-            $this->roles[] = $role;
+            $this->roles[] = clone $role;
```

Comment:
@ro0NL Because in the `serialize()` method, we have `$this->roles` which is an array, not an object (cannot be cloned).